Download.Mvc/Client/components/SearchBar.tsx

```diff
@@ -0,0 +1,31 @@
+﻿import * as React from 'react';
+import { Link } from 'react-router-dom';
+
+export class SearchBar extends React.Component<any, any>{
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            query: ''
+        }
+    }
+    handleChange = (e) => {
+        this.setState({ query: e.target.value });
+    }
+
+    handleSubmit = (e) => {
+        window.location.href = "../../search/" + encodeURIComponent(this.state.query);
+    }
+    render() {
+        return (
+            <form>
+                <input type="text" value={this.state.query} placeholder="Search . . ." onChange={this.handleChange} />
+                <Link to={`../../search/${this.state.query}`}>
+                    <button type="submit" value="Submit">
```

Comment:
@laholstege do you even need a button, form and onSubmit callback here?  I would think you could just have your textbox update your <Link> component, and that component will take them to the right search page, no window.location.href needed